### PR TITLE
Replace Compose Destinations with Safer Navigation Compose

### PR DESCRIPTION
**Pull request by uragiristereo**
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

Replace the magic navigation library with a better one: [Safer Navigation Compose](https://github.com/uragiristereo/safer-navigation-compose).

**Summary of changes:**

1. Added the required dependencies.
2. Replaced the navigation implementation.
3. Purged Compose Destinations leftovers.

**Tested changes:**

The app is running perfectly.

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -86,9 +83,11 @@ dependencies {
 
     coreLibraryDesugaring(libs.android.desugaring)
 
-    // Compose Destinations
-    implementation(libs.compose.destinations)
-    ksp(libs.compose.destinations.ksp)
+    // Safer Navigation Compose
+    implementation(libs.safer.navigation.compose.animation)
```
Comment (boswelja) [inline app/build.gradle.kts:87]: libs.compose.saferNavigation might read nicer

File: settings.gradle.kts [hunk 2]
```diff
@@ -14,6 +14,7 @@ dependencyResolutionManagement {
     repositories {
         google()
         mavenCentral()
+        maven { setUrl("https://jitpack.io") }
```
Comment (boswelja) [inline settings.gradle.kts:17]: Where central

File: app/build.gradle.kts (hunk 1 above)
Comment (uragiristereo) [reply app/build.gradle.kts:87]: i'm just following the current convention

File: settings.gradle.kts (hunk 2 above)
Comment (uragiristereo) [reply settings.gradle.kts:17]: no

Comment (boswelja) [reply settings.gradle.kts:17]: Downvote because no central

File: app/build.gradle.kts (hunk 1 above)
Comment (boswelja) [reply app/build.gradle.kts:87]: We don't have a fixed convention for this

Comment (imashnake0) [reply app/build.gradle.kts:87]: We do actually, the strategy I use is to only create an accessor if the one before it is reused. So if we don't have a `libs.safer.*` then use camelCase.

**Comment by imashnake0** [comment]
https://discord.com/channels/1043973552904359966/1043973554468823224/1194282498306347129

**Pull request closed without merging**